### ISSUE TITLE
Rancher registry credentials not updating.

### ISSUE THREAD

**Issue by jaygorrell** [objectpartners/rancher-ecr-credentials#4]
It's possible I messed something up when tweaking things, but I have this image working in multiple environments and cannot seem to get the latest versions working.

I've deployed this image to each instance and the logs for the container regularly outputs "Successfuly update credentials 1c21 for registry 1sp8", but no matter how much I try, I'm unable to deploy an ECR image anywhere. It looks like there were some changes in the URL matching so I've tried every combination I can think of: URL, URL/repo/name, http://URL/, etc.

Any thoughts? I'm not sure what else to provide but I'd be happy to answer any questions.

For what it's worth, I do see the "Last Used" timestamp being updated in my IAM user, so that part is working fine.

**Comment by johnrengelman** [comment]
What do you have configured for the Registry address in Rancher?

**Comment by johnrengelman** [comment]
Also, can you check the API Audit logs in Rancher to see if you can see a `api.registryCredential.update` call?

**Comment by johnrengelman** [comment]
@jaygorrell I found a couple issues in the code and released a version 1.0.1 of the container, can you try that one and let me know if it works for you?

**Comment by jaygorrell** [comment]
@johnrengelman That did it! Just curious, was there something unintended I did that was preventing the previous version from working?

**Comment by johnrengelman** [comment]
Well, it depends on how you enter the server address into Rancher...I had changed the code to parse it as a URL because on a system I had entered it in as `https://registry.foo.com` and then it didn't match the value that was coming from AWS.
But in doing that, it broke if you if entered the value in Rancher as just `registry.foo.com`. because the URL parsing in Go ends up putting that value into the `Path` value instead of `Host`.

So I just had to handle that situation.
